### PR TITLE
EIP150 version1 gasPrice change proposal

### PR DESCRIPTION
**Pull request by winsvega**
Implements http://github.com/ethereum/EIPs/issues/150

 - [x] changes in gas costs
 - [ ] define transition block for the main network
 - [ ] the 63/64 rule for limiting the call stack
 - [x] test for the gas cost changes
 - [ ] test for 63/64 call gas rules

### CODE REVIEW

File: libevm/VM.cpp [hunk 1]
```diff
@@ -543,6 +544,7 @@ void VM::interpretCases()
 
 		CASE_BEGIN(BALANCE)
 		{
+			m_runGas += toUint64(m_schedule->balanceGas);
```
Comment (chfast) [inline libevm/VM.cpp]: It previously was charged ~20 gas. Are you sure it will not be now 20 + balanceGas?

Comment (winsvega) [inline libevm/VM.cpp]: I am not. 
should we null it in gas prices list?

Comment (chriseth) [inline libevm/VM.cpp]: You should change the schedule itself, not just add something to rungas here manually.

File: libethashseal/Ethash.cpp [hunk 2]
```diff
@@ -84,6 +84,9 @@ StringHashMap Ethash::jsInfo(BlockHeader const& _bi) const
 
 EVMSchedule const& Ethash::evmSchedule(EnvInfo const& _envInfo) const
 {
+	if (_envInfo.number() >= chainParams().u256Param("devconforkBlock"))
+		return DevconSchedule;
+	else
```
Comment (chriseth) [inline libethashseal/Ethash.cpp]: Style

File: libethashseal/GenesisInfo.cpp [hunk 3]
```diff
@@ -159,6 +160,7 @@ R"E(
 		"accountStartNonce": "0x00",
 		"frontierCompatibilityModeLimit": "0x05",
 		"daoHardforkBlock": "0x08",
+		"devconforkBlock": "0x0a",
```
Comment (chriseth) [inline libethashseal/GenesisInfo.cpp]: Is there enough space between `0x08` and `0x0a` to perform all necessary tests?

File: libevmcore/EVMSchedule.h [hunk 4]
```diff
@@ -63,11 +63,33 @@ struct EVMSchedule
...
+	//EIP150
+	unsigned extcodesizeGas = 20;
+	unsigned extcodecopyGas = 20;
+	unsigned balanceGas = 20;
+	unsigned suicideGas = 0;
+
 };
 
 static const EVMSchedule DefaultSchedule = EVMSchedule();
+static EVMSchedule EVMScheduleDevcon()
```
Comment (chriseth) [inline libevmcore/EVMSchedule.h]: Please call it `EVMScheduleEIP150`

File: libevmcore/EVMSchedule.h [hunk 5]
```diff
@@ -63,11 +63,33 @@ struct EVMSchedule
...
+{
+	//EIP150
+	EVMSchedule m_schedule = DefaultSchedule;
```
Comment (chriseth) [inline libevmcore/EVMSchedule.h]: Shouldn't this be a modified HomesteadSchedule?

File: libevmcore/Instruction.cpp [hunk 6]
```diff
@@ -195,7 +195,7 @@ static const std::map<Instruction, InstructionInfo> c_instructionInfo =
 	{ Instruction::SIGNEXTEND,  { "SIGNEXTEND",     0, 2, 1, false, LowTier } },
 	{ Instruction::SHA3,        { "SHA3",           0, 2, 1, false, SpecialTier } },
 	{ Instruction::ADDRESS,     { "ADDRESS",        0, 0, 1, false, BaseTier } },
-	{ Instruction::BALANCE,     { "BALANCE",        0, 1, 1, false, ExtTier } },
+	{ Instruction::BALANCE,     { "BALANCE",        0, 1, 1, false, ZeroTier } },
```
Comment (chriseth) [inline libevmcore/Instruction.cpp]: It should be SpecialTier (also the others)

File: test/libethereum/BlockchainTestsFiller/TestNetwork/bcEIP150TestFiller.json [hunk 7]
```diff
@@ -0,0 +1,335 @@
...
+	    },
+	    "1000000000000000000000000000000000000012" : {
+		"shoulsnotexist" : "1"
+	    },
+	    "1000000000000000000000000000000000000013" : {
+		"shoulsnotexist" : "1"
+	    }
+        },
+        "pre" : {
+            "d4fe7bc31cedb7bfb8a345f31e668033056b2728" : { "balance" : "1000000", "nonce" : "0", "code" : "", "storage": {} },
```
Comment (chriseth) [inline test/libethereum/BlockchainTestsFiller/TestNetwork/bcEIP150TestFiller.json]: Is this the attack contract? If yes, I don't think this is a good test. What we need to test here is gas prices, not resource usage.

File: libevm/VM.cpp (hunk 1 above)
Comment (chriseth) [inline libevm/VM.cpp]: Change it to how it is done for e.g. sha3, i.e. leave it like it is here and change the tier to specialTier.

File: test/libethereum/BlockchainTestsFiller/TestNetwork/bcEIP150TestFiller.json (hunk 7 above)
Comment (winsvega) [inline test/libethereum/BlockchainTestsFiller/TestNetwork/bcEIP150TestFiller.json]: this is a contract for dao transition. have to carry it with test for the transition rules

File: libethashseal/Ethash.cpp (hunk 2 above)
Comment (winsvega) [inline libethashseal/Ethash.cpp]: fixed with https://github.com/ethereum/cpp-ethereum/pull/3344/commits/7a39c97071b55cbeb09fb71bc5d7b8a336586cf4

**Pull request closed without merging**